### ISSUE TITLE
Applying SSD to Spacenet data (repo available to suggest improvements)

### ISSUE THREAD

**Issue by aurotripathy** [weiliu89/caffe#210]
Hi, 
This project (link below) applies the Caffe-based Single-Shot Detector (SSD) algorithm to the [Spacenet](https://aws.amazon.com/public-data-sets/spacenet/) dataset. Early results (demonstrating feasibility) are shown below.

https://github.com/aurotripathy/ssd-spacenet

Looks promising, but mAP is still quite low (0.3) after 17,000 iterations and an (effective) learning rate of 0.0001. In the training script, the `base_lr` is 0.000004.

If you have questions, I'll gladly answer them.

Suggestions for applying image prepossessing, image data augmentation,  and hyperparameter selection are _very_ welcome.

**Comment by weiliu89** [comment]
Why is the learning rate so low? Are you fine-tuning from one of the provided models?

**Comment by aurotripathy** [comment]
@weiliu89 Thanks for taking a look. 
Yes, I'm fine-tuning from exactly the same model as the PASCAL VOC dataset, `VGG_ILSVRC_16_layers_fc_reduced.caffemodel`.

Any guidance on whether the is appropriate for satellite imagery datasets?

**Comment by weiliu89** [comment]
Why not use same base_lr (0.001)? If the loss goes to nan after some iterations, you could use lower lr to train for a few hundred iterations,  cancel the job after the loss is stable, and resume training with 0.001.

0.0001 seems too low unless you are fine-tuning from a trained VOC or COCO model.

**Comment by aurotripathy** [comment]
What you suggest is interesting and I will try it that way. 
I do recall the loss going to NaN right away (after just a few iterations)
